### PR TITLE
build, release-notes.md: Document 0.10.0 release

### PR DESCRIPTION
**Pull request by rbradford**
Update release notes and version number for the new release.

Signed-off-by: Rob Bradford <robert.bradford@intel.com>

### CODE REVIEW

**Comment by rbradford** [comment]
@sameo @sboeuf I went through all the merged PRs. The list seems a bit shorter this time. There were a few more less user visible changes this cycle like integration test improvements and virtio device refactoring.

File: release-notes.md [hunk 1]
```diff
@@ -90,6 +88,52 @@
...
 
+# v0.10.0
+
+This release has been tracked through the [0.10.0 project](https://github.com/cloud-hypervisor/cloud-hypervisor/projects/13).
+
+Highlights for `cloud-hypervisor` version 0.10.0 include:
+
+### `virtio-block` Support for Multiple Descriptors
+
+Some `virtio-block` device drivers may generate requests with multiple descriptors. Support has been added for this however there are still some issues when using the non-default `io_uring` feature (#1715).
```
Comment (sboeuf) [inline release-notes.md:99]: The io_uring issue is now solved.

File: release-notes.md [hunk 2]
```diff
@@ -90,6 +88,52 @@
...
+
+### Memory Zones
+
+Support has been added for fine grained control of memory allocation for the guest. This includes controlling the backing of sections of guest memory, assigning to specific host NUMA nodes and assigning memory and vCPUs to specific memory nodes inside the guest. Full details of this can be found in the [memory documentation](docs/memory.md).
+
```
Comment (sboeuf) [inline release-notes.md:104]: s/specific memory nodes/specific NUMA nodes/

File: release-notes.md [hunk 3]
```diff
@@ -90,6 +88,52 @@
...
+### `Seccomp` Sandbox Improvements
+
+All the remaining threads and devices are now isolated within their own `seccomp` filters. This provides a layer of sandboxing and enhances the security model of `cloud-hypervisor`.
+
+### Preliminary KVM HyperV Emulation Control
+
+A new option (`kvm_hyperv`) has been added to `--cpus` to provide an option to toggle on KVM's HyperV emulation support. This enables progress towards booting Windows without adding extra emulated devices.
+
```
Comment (sboeuf) [inline release-notes.md:112]: Do we want to mention that we have SGX CI in place now?

And also maybe that we did a bunch of refactoring during this release?

**Review by sboeuf — APPROVED** [review]
Looking good! Feel free to merge after addressing the comments.

**Review by likebreath — APPROVED** [review]
Yay! Go CH v0.10.0.

File: release-notes.md (hunk 3 above)
Comment (rbradford) [reply release-notes.md:112]: I don't think so. The release notes are about what is visible to the user - those two things are not.

Comment (sboeuf) [reply release-notes.md:112]: Fair enough :)

**Review by sameo — APPROVED** [review]
Quite a release...LGTM

**Pull request merged**